core/ledger/pvtdatastorage/store_created_from_snapshot_test.go

```diff
@@ -0,0 +1,327 @@
+		store := setup(
+			[]*snapshotData{
+				{
+					collection: "eligible-coll",
+					keyHash:    []byte("eligible-coll-key-hash"),
+					valueHash:  []byte("eligible-coll-value-hash"),
+					version:    version.NewHeight(20, 200),
+				},
+				{
+					collection: "ineligible-coll",
+					keyHash:    []byte("ineligible-coll-key-hash"),
+					valueHash:  []byte("ineligible-coll-value-hash"),
+					version:    version.NewHeight(21, 210),
+				},
+				{
+					collection: "eligible-coll",
+					keyHash:    []byte("key-hash-at-last-block-in-snapshot"),
+					valueHash:  []byte("value-hash-at-last-block-in-snapshot"),
+					version:    version.NewHeight(25, 250),
+				},
+			},
+		)
+
+		m, err := store.FetchBootKVHashes(20, 200, "ns", "eligible-coll")
+		require.NoError(t, err)
+		require.Equal(t,
+			map[string][]byte{
+				"eligible-coll-key-hash": []byte("eligible-coll-value-hash"),
+			},
+			m,
+		)
+
+		m, err = store.FetchBootKVHashes(25, 250, "ns", "eligible-coll")
+		require.NoError(t, err)
+		require.Equal(t,
+			map[string][]byte{
+				"key-hash-at-last-block-in-snapshot": []byte("value-hash-at-last-block-in-snapshot"),
+			},
+			m,
+		)
+
+		m, err = store.FetchBootKVHashes(21, 210, "ns", "ineligible-coll")
+		require.NoError(t, err)
+		require.Equal(t,
+			map[string][]byte{
+				"ineligible-coll-key-hash": []byte("ineligible-coll-value-hash"),
+			},
+			m,
+		)
+
+		missingDataInfo, err := store.GetMissingPvtDataInfoForMostRecentBlocks(4)
+		require.NoError(t, err)
+		require.Equal(
+			t,
+			ledger.MissingPvtDataInfo{
+				20: ledger.MissingBlockPvtdataInfo{
+					200: []*ledger.MissingCollectionPvtDataInfo{
+						{
+							Namespace:  "ns",
+							Collection: "eligible-coll",
+						},
+					},
+				},
+				25: ledger.MissingBlockPvtdataInfo{
+					250: []*ledger.MissingCollectionPvtDataInfo{
+						{
+							Namespace:  "ns",
+							Collection: "eligible-coll",
+						},
+					},
+				},
+			},
+			missingDataInfo,
+		)
+
+		err = store.CommitPvtDataOfOldBlocks(
+			map[uint64][]*ledger.TxPvtData{
+				20: {
+					produceSamplePvtdata(t, 200, []string{"ns:eligible-coll"}),
+				},
+				25: {
+					produceSamplePvtdata(t, 250, []string{"ns:eligible-coll"}),
+				},
+			},
+			nil,
+		)
+		require.NoError(t, err)
+
+		missingDataInfo, err = store.GetMissingPvtDataInfoForMostRecentBlocks(4)
+		require.NoError(t, err)
+		require.Len(t, missingDataInfo, 0)
+
+		m, err = store.FetchBootKVHashes(20, 200, "ns", "eligible-coll")
+		require.NoError(t, err)
+		require.Len(t, m, 0)
+
+		m, err = store.FetchBootKVHashes(25, 250, "ns", "eligible-coll")
+		require.NoError(t, err)
+		require.Len(t, m, 0)
+
+		m, err = store.FetchBootKVHashes(21, 210, "ns", "ineligible-coll")
+		require.NoError(t, err)
+		require.Equal(t,
+			map[string][]byte{
+				"ineligible-coll-key-hash": []byte("ineligible-coll-value-hash"),
+			},
+			m,
+		)
+	})
+
+	t.Run("purger-deletes-expired-bootKV-hashes", func(t *testing.T) {
```

Comment:
Existing test for purger is present in `TestStorePurge()` (in the alternative file `store_test.go`) which takes care of deleting eligible, ineligible, and depriritized missing data along with the expiry entry. IMO, it would be good to have all purge related testing in a single place so that we would know where to look for a particular test.